File: dev-docs/snippets/documents/delete.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Delete a document from the server and evict its local data.
export async function deleteDocument(client: JsBaoClient, documentId: string) {
  // #region example
  await client.documents.delete(documentId, { forceCloseIfOpen: true });
  // #endregion example
}

File: dev-docs/snippets/documents/delete.swift
import JsBaoClient

// Delete a document. Swift returns an untyped `[String: Any]` (vs JS `void`)
// and does not perform local eviction.
func deleteDocument(client: JsBaoClient, documentId: String) async throws {
  // #region example
  let result = try await client.documents.delete(
    documentId: documentId, forceCloseIfOpen: true
  )
  // #endregion example
  _ = result
}
